packages/core-react/src/components/TextField/index.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+
+export default function TextField({
+  id,
+  type,
+  placeholder,
+  disabled,
+  required,
+  value,
+  error,
+  compact,
+  label,
+  ...rest
+}) {
+  const wrapperClass = clsx('ray-text-field', {
+    'ray-text-field--has-value': value,
+    'ray-text-field--error': error,
+    'ray-text-field--disabled': disabled,
+    'ray-text-field--required': required,
+    'ray-text-field--compact': compact
+  });
+
+  return (
+    <div className={wrapperClass}>
+      <input
+        className="ray-text-field__input"
+        id={id}
+        type={type}
+        value={value}
+        placeholder={placeholder}
+        disabled={disabled}
+        {...rest}
+      />
+      {label && (
+        <label className="ray-text-field__label" htmlFor={id}>
+          {label}
+        </label>
+      )}
+    </div>
```

Comment:
should we also include the error message?

Reply:
i was considering it... this pr is still wip